tests/lib/rules/callback-return.js

```diff
@@ -85,6 +86,40 @@ ruleTester.run("callback-return", rule, {
             options: [["cb", "next"]]
         },
 
+        // allow object methods (https://github.com/eslint/eslint/issues/4711)
+        {
+            code: "function a(err) { if (err) { callback(err); } }",
+            options: [["obj.method"]]
+        },
+        {
+            code: "function a(err) { if (err) { return obj.method(err); } }",
+            options: [["obj.method"]]
+        },
+        {
+            code: "function a(err) { if (err) { obj.method(err); } }",
+            options: [["obj().method"]]
+        },
+        {
+            code: "function a(err) { if (err) { //comment\nreturn obj().method(err); } }",
+            options: [["obj().method"]]
+        },
```

Comment:
I think so, I'd just say we need a member expression whose left side is an identifier, otherwise, don't do anything.